lib/open_api_spex/path_item.ex

```diff
@@ -69,6 +69,10 @@ defmodule OpenApiSpex.PathItem do
     |> from_valid_routes()
   end
 
+  def empty?(%__MODULE__{} = path_item) do
```

Comment:
[suggestion] Instead of introducing a public function (which we should ideally document and add a typespec for), we can fix the root of the problem with:

```diff
diff --git a/lib/open_api_spex/path_item.ex b/lib/open_api_spex/path_item.ex
index 584774f..d41a913 100644
--- a/lib/open_api_spex/path_item.ex
+++ b/lib/open_api_spex/path_item.ex
@@ -77,16 +77,17 @@ defmodule OpenApiSpex.PathItem do
   defp from_valid_routes([]), do: nil

   defp from_valid_routes(routes) do
-    attrs =
-      routes
-      |> Enum.map(fn route ->
-        case Operation.from_route(route) do
-          nil -> nil
-          op -> {route.verb, op}
-        end
-      end)
-      |> Enum.filter(& &1)
-
-    struct(PathItem, attrs)
+    routes
+    |> Enum.map(fn route ->
+      case Operation.from_route(route) do
+        nil -> nil
+        op -> {route.verb, op}
+      end
+    end)
+    |> Enum.filter(& &1)
+    |> case do
+      [] -> nil
+      attrs -> struct(PathItem, attrs)
+    end
   end
 end
```

`from_valid_routes/1` should be returning `nil` when there are no valid routes instead of a default struct. This way we don't need to modify `OpenApiSpex.Paths`.

Reply:
This is way better. Addressed in 711f2ba